Work branch: Guard middleware

erastian opened the pull request:
- added guard middleware
- updated articles, auth, categories and users controllers for using guard middleware
- added "Publish" method to articles module
- Added "url" field in Articles model
- create and update DTOs was updated
- Updated Articles module with a lot of internal methods

TODO: Comments module still in another branch. Need to come back to this module after comments module merge in main branch. Article module needs a revision of the methods

erastian committed:
Work branch: Guard middleware

- added guard middleware
- updated articles, auth, categories and users controllers for using guard middleware
- added "Publish" method to articles module

TODO: comments module still in another branch. Need to come back to this module after comments module merge in main branch

mdziarkach commented on src/articles/articles.controller.ts:
```diff
@@ -26,15 +30,35 @@ export class ArticlesController extends BaseController implements IArticlesContr
+				method: 'post',
+				func: this.publishArticle,
+				middlewares: [
+					new GuardMiddleware([Role.ADMIN, Role.MANAGER]),
```
only admin

erastian replied:
Done

mdziarkach commented on src/articles/articles.controller.ts:
```diff
@@ -78,6 +102,19 @@ export class ArticlesController extends BaseController implements IArticlesContr
+
 	async deleteArticle({ params }: Request, res: Response, next: NextFunction): Promise<void> {
 		try {
 			const id = params.id;
```
if manager didn't create this article then he can't delete it

erastian replied:
Done

mdziarkach commented on src/articles/articles.controller.ts:
```diff
@@ -26,15 +30,35 @@ export class ArticlesController extends BaseController implements IArticlesContr
 				func: this.updateArticle,
-				middlewares: [new ValidateMiddleware(ArticleUpdateDto)],
+				middlewares: [
+					new GuardMiddleware([Role.ADMIN, Role.MANAGER]),
```
manager can update only own article

erastian replied:
Done

mdziarkach commented on src/common/guard.middleware.ts:
```diff
@@ -0,0 +1,22 @@
+
+	execute(req: Request, res: Response, next: NextFunction): void {
+		if (!req.user) {
+			return next(new HTTPError(StatusCodes.FORBIDDEN, 'User not authorized'));
```
let's return 401 status code here

erastian replied:
Done

erastian committed:
Work branch: Guard middleware

- Added "url" field in Articles model
- create and update DTOs was updated
- Updated Articles module with a lot of internal methods

erastian committed:
Work branch: Guard middleware

- create & update DTOs update with @IsMongoId
